src/riak_kv.proto

```diff
@@ -365,6 +366,22 @@ message RpbFetchResp{
     optional uint32 crc_check = 3; // CRC check of encoded object
     optional bytes replencoded_object = 4; // Special repl encoding using for object 
     optional bytes deleted_vclock = 5;
+    optional uint32 segment_id = 6;
+    optional uint32 segment_hash = 7; 
+}
+
+// Push request - queue a list of keys for replication if at a given clock.
+message RpbPushReq{
+    required bytes queuename = 1;
+    repeated RpbKeysValue keys_value = 2;
+}
+
+message RpbPushResp{
+    required bool queue_exists = 1;
+    optional bytes queuename = 2;
```

Comment:
IN all above, queuename = 1.. this does not matter, of course, but it would be more consistent to turn name and exist around